services/streammanager/kafka/client/producer.go

```diff
@@ -140,28 +138,45 @@ func headers(msg Message) (headers []kafka.Header) {
 }
 
 func isErrTemporary(err error) bool {
+	if errors.Is(err, context.DeadlineExceeded) {
+		return true
+	}
 	isTransientNetworkError := errors.Is(err, io.ErrUnexpectedEOF) ||
 		errors.Is(err, syscall.ECONNREFUSED) ||
 		errors.Is(err, syscall.ECONNRESET) ||
 		errors.Is(err, syscall.EPIPE)
 	if isTransientNetworkError {
 		return true
 	}
+	var tempError interface{ Temporary() bool }
 	if errors.As(err, &tempError) {
 		return tempError.Temporary()
 	}
```

Comment:
BTW, I believe that the language has moved away from this `Temporary` error pattern, at least according to [that](https://go-review.googlesource.com/c/go/+/188398)

Reply:
Errors from the underlying segment library still follow this pattern so we might have to stick with it for now.